### ISSUE TITLE
Multiple mobs logged for consumption but are not consumed.

### ISSUE THREAD

**Issue by LaneBren** [Xian55/WowClassicGrindBot#431]
### Description

Using modified DeathKnight_80_Frost.json (deleted higher level spells) on level 58 DK in Hellfire with recorded path.  After entering combat with multiple enemies, only last enemy is looted. Log shows multiple corpses are recorded for consumption, but seems like it cannot target corpses for consumption and gives up. 

Not set to KeyboardOnly.

Keybinds seem to be set correctly if the consumption functionality tries to cycle through previous targets: see screenshots

### Addon Version

1.7.0

### World of Warcraft Client

Wrath of the Lich King Classic

### Reproduction Steps

1. Enter Combat
2. Leave combat with with more than 1 mob
3. Only last mob that is killed is consumed
4.

### Last Good Version

1.7.0

### Screenshots

![image](https://user-images.githubusercontent.com/25021468/193182792-1af6d6ed-8d65-41a4-aea5-940778946713.png)

### Logs

<html>
<body>

00:08:59 | Inactive, kill credit detected!
-- | --
00:08:52 | Inactive, kill credit detected!
00:08:47 | ["Death Grip"] GCD: 0ms \| Cast: 0ms \| Next spell -354ms
00:08:47 | ["Death Grip"] ... AfterCastWaitMeleeRange
00:08:47 | ["Death Grip"] ... AfterCastWaitCombat ? True 0.0001ms
00:08:47 | ["Death Grip"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:47 | ["Death Grip"] instant input 80ms True 121.4681ms
00:08:47 | Stop auto interact!
00:08:47 | New Plan= " Pull Target"
00:08:44 | Stick to initial target!
00:08:43 | New Plan= " Approach Target"
00:08:43 | [NpcNameFinder] type = "None"
00:08:43 | [NpcNameFinder] type = "None"
00:08:42 | Pathfinder - 118.29238 - "<-456.99573, 1734.9431, 37.428104>" -> "<-497.1294, 1846.2192, 37.428104>" 13.0261ms
00:08:42 | New Plan= " Follow Route"
00:08:42 | ----- Corpse consumed. Total: 0 \| Remaining: 0
00:08:42 | New Plan= " Corpse Consumed"
00:08:42 | LootGoal: Loot Failed, target not found!
00:08:41 | Look at possible closest corpse and try once again...
00:08:41 | New Plan= " Loot"
00:08:41 | ----- Safe to consume a corpse.
00:08:41 | New Plan= " Consume Corpse"
00:08:41 | ----- Corpse consumed. Total: 4 \| Remaining: 1
00:08:41 | New Plan= " Corpse Consumed"
00:08:41 | LootGoal: Loot Failed, target not found!
00:08:40 | Look at possible closest corpse and try once again...
00:08:40 | New Plan= " Loot"
00:08:40 | ----- Safe to consume a corpse.
00:08:40 | New Plan= " Consume Corpse"
00:08:40 | ----- Corpse consumed. Total: 4 \| Remaining: 2
00:08:40 | New Plan= " Corpse Consumed"
00:08:40 | LootGoal: Loot Successful after 855.5325ms
00:08:39 | LootGoal: Target Last Target Found!
00:08:39 | Look at possible closest corpse and try once again...
00:08:38 | New Plan= " Loot"
00:08:38 | ----- Safe to consume a corpse.
00:08:38 | New Plan= " Consume Corpse"
00:08:38 | ----- Corpse consumed. Total: 4 \| Remaining: 3
00:08:38 | New Plan= " Corpse Consumed"
00:08:38 | LootGoal: Loot Failed, target not found!
00:08:38 | Look at possible closest corpse and try once again...
00:08:37 | [NpcNameFinder] type = "Corpse"
00:08:37 | New Plan= " Loot"
00:08:37 | ----- Safe to consume a corpse.
00:08:37 | New Plan= " Consume Corpse"
00:08:37 | Lost target!
00:08:37 | ["Icy Touch"] instant input 72ms False 472.3112ms
00:08:37 | ["Icy Touch"] instant usable: True->True -- "NONE"->"SPELL_FAILED_TARGETS_DEAD"
00:08:37 | ["Icy Touch"] instant input 92ms True 130.6966ms
00:08:36 | Kill credit detected! Known kills: 4 \| Fighting with: 0
00:08:36 | ["Trinket 1"] GCD: 100ms \| Cast: 0ms \| Next spell -254ms
00:08:36 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:36 | ["Trinket 1"] instant input 53ms True 0.0004ms
00:08:35 | ["Blood Strike"] GCD: 1500ms \| Cast: 0ms \| Next spell 1146ms
00:08:35 | ["Blood Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:35 | ["Blood Strike"] instant input 79ms True 120.6976ms
00:08:34 | ["Trinket 1"] GCD: 66ms \| Cast: 0ms \| Next spell -288ms
00:08:34 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:34 | ["Trinket 1"] instant input 65ms True 0.0009ms
00:08:33 | ["Icy Touch"] GCD: 1466ms \| Cast: 0ms \| Next spell 1112ms
00:08:33 | ["Icy Touch"] ... AfterCastAuraExpected 1900ms
00:08:33 | ["Icy Touch"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:33 | ["Icy Touch"] instant input 81ms True 152.4994ms
00:08:33 | ["Trinket 1"] GCD: 0ms \| Cast: 0ms \| Next spell -354ms
00:08:33 | ["Trinket 1"] instant usable: True->False -- "NONE"->"NONE"
00:08:33 | ["Trinket 1"] instant input 90ms True 0.0004ms
00:08:32 | ["Trinket 1"] GCD: 67ms \| Cast: 0ms \| Next spell -287ms
00:08:32 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:32 | ["Trinket 1"] instant input 91ms True 0.0003ms
00:08:31 | ["Plague Strike"] GCD: 1467ms \| Cast: 0ms \| Next spell 1113ms
00:08:31 | ["Plague Strike"] ... AfterCastAuraExpected 1900ms
00:08:31 | ["Plague Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:31 | ["Plague Strike"] instant input 67ms True 149.3476ms
00:08:30 | ["Trinket 1"] GCD: 0ms \| Cast: 0ms \| Next spell -354ms
00:08:30 | ["Trinket 1"] instant usable: True->False -- "NONE"->"NONE"
00:08:30 | ["Trinket 1"] instant input 82ms True 0.0004ms
00:08:30 | ["Trinket 1"] GCD: 67ms \| Cast: 0ms \| Next spell -287ms
00:08:30 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:30 | ["Trinket 1"] instant input 96ms True 0.0011ms
00:08:28 | ["Frost Strike"] GCD: 1500ms \| Cast: 0ms \| Next spell 1146ms
00:08:28 | ["Frost Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:28 | ["Frost Strike"] instant input 98ms True 102.2198ms
00:08:28 | ["Trinket 1"] GCD: 100ms \| Cast: 0ms \| Next spell -254ms
00:08:28 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:28 | ["Trinket 1"] instant input 60ms True 0.0003ms
00:08:27 | Lost target!
00:08:27 | ["Icy Touch"] GCD: 1466ms \| Cast: 0ms \| Next spell 1112ms
00:08:27 | ["Icy Touch"] ... AfterCastAuraExpected 1900ms
00:08:27 | ["Icy Touch"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:27 | ["Icy Touch"] instant input 81ms True 136.4866ms
00:08:27 | Kill credit detected! Known kills: 3 \| Fighting with: 1
00:08:26 | ["Trinket 1"] GCD: 100ms \| Cast: 0ms \| Next spell -254ms
00:08:26 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:26 | ["Trinket 1"] instant input 62ms True 0.0009ms
00:08:25 | ["Plague Strike"] GCD: 1466ms \| Cast: 0ms \| Next spell 1112ms
00:08:25 | ["Plague Strike"] ... AfterCastAuraExpected 1900ms
00:08:25 | ["Plague Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:25 | ["Plague Strike"] instant input 55ms True 143.3164ms
00:08:24 | ["Trinket 1"] GCD: 99ms \| Cast: 0ms \| Next spell -255ms
00:08:24 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:24 | ["Trinket 1"] instant input 65ms True 0.0008ms
00:08:23 | Found new target!
00:08:23 | Checking target in front...
00:08:23 | Lost target!
00:08:23 | ["Icy Touch"] GCD: 1466ms \| Cast: 0ms \| Next spell 1112ms
00:08:23 | ["Icy Touch"] ... AfterCastAuraExpected 1900ms
00:08:23 | ["Icy Touch"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:23 | ["Icy Touch"] instant input 51ms True 150.3705ms
00:08:23 | Kill credit detected! Known kills: 2 \| Fighting with: 2
00:08:23 | ["Trinket 1"] GCD: 0ms \| Cast: 0ms \| Next spell -354ms
00:08:23 | ["Trinket 1"] instant usable: True->False -- "NONE"->"NONE"
00:08:23 | ["Trinket 1"] instant input 95ms True 0.0004ms
00:08:22 | ["Trinket 1"] GCD: 0ms \| Cast: 0ms \| Next spell -354ms
00:08:22 | ["Trinket 1"] instant usable: True->False -- "NONE"->"NONE"
00:08:22 | ["Trinket 1"] instant input 67ms True 0.0002ms
00:08:22 | ["Trinket 1"] GCD: 32ms \| Cast: 0ms \| Next spell -322ms
00:08:22 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:22 | ["Trinket 1"] instant input 100ms True 0.0001ms
00:08:20 | ["Plague Strike"] GCD: 1466ms \| Cast: 0ms \| Next spell 1112ms
00:08:20 | ["Plague Strike"] ... AfterCastAuraExpected 1900ms
00:08:20 | ["Plague Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:20 | ["Plague Strike"] instant input 86ms True 116.3586ms
00:08:20 | ["Trinket 1"] GCD: 0ms \| Cast: 0ms \| Next spell -354ms
00:08:20 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:20 | ["Trinket 1"] instant input 72ms True 0.0001ms
00:08:18 | ["Frost Strike"] GCD: 1500ms \| Cast: 0ms \| Next spell 1146ms
00:08:18 | ["Frost Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:18 | ["Frost Strike"] instant input 72ms True 27.2071ms
00:08:18 | ["Trinket 1"] GCD: 0ms \| Cast: 0ms \| Next spell -354ms
00:08:18 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:18 | ["Trinket 1"] instant input 71ms True 0.0009ms
00:08:17 | New Plan= NO PLAN
00:08:17 | Lost target!
00:08:17 | ["Blood Boil"] GCD: 1500ms \| Cast: 0ms \| Next spell 1146ms
00:08:17 | ["Blood Boil"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:17 | ["Blood Boil"] instant input 85ms True 158.5823ms
00:08:17 | Kill credit detected! Known kills: 1 \| Fighting with: 1
00:08:16 | ["Trinket 1"] GCD: 0ms \| Cast: 0ms \| Next spell -354ms
00:08:16 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:16 | ["Trinket 1"] instant input 82ms True 0.0002ms
00:08:15 | ["Plague Strike"] GCD: 1467ms \| Cast: 0ms \| Next spell 1113ms
00:08:15 | ["Plague Strike"] ... AfterCastAuraExpected 1900ms
00:08:15 | ["Plague Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:15 | ["Plague Strike"] instant input 83ms True 50.17ms
00:08:14 | ["Unbreakable Armor"] GCD: 0ms \| Cast: 0ms \| Next spell -354ms
00:08:14 | ["Unbreakable Armor"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:14 | ["Unbreakable Armor"] instant input 57ms True 142.572ms
00:08:14 | ["Trinket 1"] GCD: 0ms \| Cast: 0ms \| Next spell -354ms
00:08:14 | ["Trinket 1"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:14 | ["Trinket 1"] instant input 75ms True 0.0001ms
00:08:13 | New Plan= " Combat"
00:08:13 | ["Icy Touch"] GCD: 1466ms \| Cast: 0ms \| Next spell 1112ms
00:08:13 | ["Icy Touch"] ... AfterCastWaitCombat ? True 0ms
00:08:13 | ["Icy Touch"] ... AfterCastAuraExpected 1900ms
00:08:13 | ["Icy Touch"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
00:08:13 | ["Icy Touch"] instant input 77ms True 123.4699ms
00:08:12 | Stop auto interact!
00:08:12 | New Plan= " Pull Target"
00:08:12 | [NpcNameFinder] type = "None"
00:08:12 | ["LocalPathingApi"] distance from nearlest point is 124.52691. Have to clear RouteToWaypoint.
00:08:12 | New Plan= " Follow Route"

</body>
</html>

**Comment by Xian55** [comment]
Hello there.

Firstly, i would like to ask you to upload the altered class config file based on the `DeathKnight_80_Frost.json`.

Secondly, cherry picked log snippets can be misleading, upload the complete logfile to be able to conduct further investigation.

**Comment by LaneBren** [comment]
New full log with replicated issue.

<html>
<body>


14:19:54 | [NpcNameFinder] type = "None"
-- | --
14:19:53 | [NpcNameFinder] type = "None"
14:19:53 | ["LocalPathingApi"] distance is close 26.43027. Keep RouteToWaypoint.
14:19:53 | New Plan= " Follow Route"
14:19:53 | ----- Corpse consumed. Total: 0 \| Remaining: 0
14:19:53 | New Plan= " Corpse Consumed"
14:19:53 | LootGoal: Loot Failed, target not found!
14:19:52 | Look at possible closest corpse and try once again...
14:19:52 | New Plan= " Loot"
14:19:52 | ----- Safe to consume a corpse.
14:19:52 | New Plan= " Consume Corpse"
14:19:52 | ----- Corpse consumed. Total: 2 \| Remaining: 1
14:19:52 | New Plan= " Corpse Consumed"
14:19:52 | LootGoal: Loot Successful after 820.4242ms
14:19:51 | LootGoal: Target Last Target Found!
14:19:51 | Look at possible closest corpse and try once again...
14:19:51 | [NpcNameFinder] type = "Corpse"
14:19:51 | New Plan= " Loot"
14:19:51 | ----- Safe to consume a corpse.
14:19:51 | New Plan= " Consume Corpse"
14:19:51 | Lost target!
14:19:51 | ["Plague Strike"] GCD: 1466ms \| Cast: 0ms \| Next spell 1113ms
14:19:51 | ["Plague Strike"] ... AfterCastAuraExpected 1900ms
14:19:51 | ["Plague Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:51 | ["Plague Strike"] instant input 51ms True 147.9215ms
14:19:51 | Kill credit detected! Known kills: 2 \| Fighting with: 0
14:19:51 | ["Unbreakable Armor"] instant input 98ms False 466.682ms
14:19:50 | ["Unbreakable Armor"] instant input 87ms False 479.882ms
14:19:49 | ["Unbreakable Armor"] instant input 81ms False 465.655ms
14:19:49 | ["Unbreakable Armor"] instant input 98ms False 468.9298ms
14:19:47 | ["Plague Strike"] GCD: 1466ms \| Cast: 0ms \| Next spell 1113ms
14:19:47 | ["Plague Strike"] ... AfterCastAuraExpected 1900ms
14:19:47 | ["Plague Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:47 | ["Plague Strike"] instant input 73ms True 58.8741ms
14:19:47 | ["Unbreakable Armor"] instant input 52ms False 447.7333ms
14:19:47 | ["Unbreakable Armor"] instant input 79ms False 453.415ms
14:19:45 | ["Icy Touch"] GCD: 1466ms \| Cast: 0ms \| Next spell 1113ms
14:19:45 | ["Icy Touch"] ... AfterCastAuraExpected 1900ms
14:19:45 | ["Icy Touch"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:45 | ["Icy Touch"] instant input 76ms True 122.4261ms
14:19:45 | ["Unbreakable Armor"] instant input 87ms False 477.8103ms
14:19:44 | ["Unbreakable Armor"] instant input 80ms False 453.227ms
14:19:42 | ["Frost Strike"] GCD: 1500ms \| Cast: 0ms \| Next spell 1147ms
14:19:42 | ["Frost Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:42 | ["Frost Strike"] instant input 73ms True 143.1405ms
14:19:42 | ["Unbreakable Armor"] instant input 60ms False 473.6947ms
14:19:42 | Kill credit detected! Known kills: 1 \| Fighting with: 0
14:19:42 | ["Unbreakable Armor"] instant input 62ms False 470.5443ms
14:19:40 | ["Frost Strike"] GCD: 1500ms \| Cast: 0ms \| Next spell 1147ms
14:19:40 | ["Frost Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:40 | ["Frost Strike"] instant input 55ms True 144.2957ms
14:19:40 | ["Unbreakable Armor"] instant input 86ms False 447.2917ms
14:19:39 | ["Unbreakable Armor"] instant input 94ms False 472.9989ms
14:19:39 | New Plan= " Combat"
14:19:39 | ["Death Grip"] GCD: 0ms \| Cast: 0ms \| Next spell -353ms
14:19:36 | ["Death Grip"] ... AfterCastWaitMeleeRange
14:19:36 | ["Death Grip"] ... AfterCastWaitCombat ? False 3009.6989ms
14:19:33 | ["Death Grip"] instant usable: True->True -- "NONE"->"CAST_SUCCESS"
14:19:33 | ["Death Grip"] instant input 89ms True 0.0004ms
14:19:33 | ["Death Grip"] instant input 60ms False 471.9469ms
14:19:32 | Stop auto interact!
14:19:32 | New Plan= " Pull Target"
14:19:32 | New Plan= " Approach Target"
14:19:32 | ["Death Grip"] GCD: 0ms \| Cast: 0ms \| Next spell -353ms
14:19:30 | ["Death Grip"] ... AfterCastWaitMeleeRange
14:19:30 | ["Death Grip"] ... AfterCastWaitCombat ? False 3013.7651ms
14:19:27 | ["Death Grip"] instant usable: True->True -- "NONE"->"CAST_SUCCESS"
14:19:27 | ["Death Grip"] instant input 86ms True 0.0005ms
14:19:27 | ["Death Grip"] instant input 99ms False 468.2395ms
14:19:26 | Stop auto interact!
14:19:26 | New Plan= " Pull Target"
14:19:26 | [NpcNameFinder] type = "None"
14:19:26 | ["LocalPathingApi"] distance is close 9.72638. Keep RouteToWaypoint.
14:19:26 | New Plan= " Follow Route"
14:19:26 | ----- Corpse consumed. Total: 0 \| Remaining: 0
14:19:26 | New Plan= " Corpse Consumed"
14:19:26 | LootGoal: Loot Failed, target not found!
14:19:26 | Look at possible closest corpse and try once again...
14:19:25 | New Plan= " Loot"
14:19:25 | ----- Safe to consume a corpse.
14:19:25 | New Plan= " Consume Corpse"
14:19:24 | ----- Corpse consumed. Total: 4 \| Remaining: 1
14:19:24 | New Plan= " Corpse Consumed"
14:19:24 | LootGoal: Loot Failed, target not found!
14:19:24 | Look at possible closest corpse and try once again...
14:19:24 | New Plan= " Loot"
14:19:24 | ----- Safe to consume a corpse.
14:19:24 | New Plan= " Consume Corpse"
14:19:23 | ----- Corpse consumed. Total: 4 \| Remaining: 2
14:19:23 | New Plan= " Corpse Consumed"
14:19:23 | LootGoal: Loot Failed, target not found!
14:19:23 | Look at possible closest corpse and try once again...
14:19:23 | New Plan= " Loot"
14:19:23 | ----- Safe to consume a corpse.
14:19:23 | New Plan= " Consume Corpse"
14:19:23 | ----- Corpse consumed. Total: 4 \| Remaining: 3
14:19:23 | New Plan= " Corpse Consumed"
14:19:23 | LootGoal: Loot Successful after 875.4042ms
14:19:22 | LootGoal: Target Last Target Found!
14:19:22 | Look at possible closest corpse and try once again...
14:19:21 | [NpcNameFinder] type = "Corpse"
14:19:21 | New Plan= " Loot"
14:19:21 | ----- Safe to consume a corpse.
14:19:21 | New Plan= " Consume Corpse"
14:19:20 | New Plan= NO PLAN
14:19:20 | Lost target!
14:19:20 | Kill credit detected! Known kills: 4 \| Fighting with: 1
14:19:19 | ["Blood Boil"] GCD: 1500ms \| Cast: 0ms \| Next spell 1147ms
14:19:19 | ["Blood Boil"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:19 | ["Blood Boil"] instant input 97ms True 84.7044ms
14:19:19 | ["Unbreakable Armor"] instant input 97ms False 451.8002ms
14:19:19 | ["Unbreakable Armor"] instant input 99ms False 467.6142ms
14:19:17 | ["Blood Boil"] GCD: 1500ms \| Cast: 0ms \| Next spell 1147ms
14:19:17 | ["Blood Boil"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:17 | ["Blood Boil"] instant input 102ms True 99.0488ms
14:19:15 | ["Plague Strike"] GCD: 1467ms \| Cast: 0ms \| Next spell 1114ms
14:19:15 | ["Plague Strike"] ... AfterCastAuraExpected 1900ms
14:19:15 | ["Plague Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:15 | ["Plague Strike"] instant input 87ms True 107.066ms
14:19:14 | ["Icy Touch"] GCD: 1466ms \| Cast: 0ms \| Next spell 1113ms
14:19:14 | ["Icy Touch"] ... AfterCastAuraExpected 1900ms
14:19:14 | ["Icy Touch"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:14 | ["Icy Touch"] instant input 61ms True 130.3161ms
14:19:14 | ["Unbreakable Armor"] instant input 71ms False 460.8529ms
14:19:13 | ["Unbreakable Armor"] instant input 90ms False 476.1959ms
14:19:13 | ["Mind Freeze"] instant input 61ms False 470.866ms
14:19:12 | ["Mind Freeze"] instant input 62ms False 467.1417ms
14:19:10 | Found new target!
14:19:10 | Checking target in front...
14:19:10 | Lost target!
14:19:10 | ["Icy Touch"] GCD: 1466ms \| Cast: 0ms \| Next spell 1113ms
14:19:10 | ["Icy Touch"] ... AfterCastAuraExpected 1900ms
14:19:10 | ["Icy Touch"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:10 | ["Icy Touch"] instant input 72ms True 128.1925ms
14:19:10 | Kill credit detected! Known kills: 3 \| Fighting with: 2
14:19:10 | ["Unbreakable Armor"] instant input 55ms False 477.0914ms
14:19:10 | ["Unbreakable Armor"] instant input 51ms False 448.5458ms
14:19:09 | Kill credit detected! Known kills: 2 \| Fighting with: 3
14:19:08 | ["Blood Boil"] GCD: 1500ms \| Cast: 0ms \| Next spell 1147ms
14:19:08 | ["Blood Boil"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:08 | ["Blood Boil"] instant input 51ms True 114.1074ms
14:19:08 | ["Unbreakable Armor"] instant input 98ms False 468.1437ms
14:19:07 | ["Unbreakable Armor"] instant input 99ms False 467.1694ms
14:19:07 | ["Pestilenc"] instant input 96ms False 451.4972ms
14:19:06 | ["Pestilenc"] instant input 97ms False 452.3439ms
14:19:05 | ["Mind Freeze"] instant input 63ms False 469.1032ms
14:19:05 | ["Mind Freeze"] instant input 75ms False 458.5811ms
14:19:03 | ["Plague Strike"] GCD: 1466ms \| Cast: 0ms \| Next spell 1113ms
14:19:03 | ["Plague Strike"] ... AfterCastAuraExpected 1900ms
14:19:03 | ["Plague Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:03 | ["Plague Strike"] instant input 58ms True 41.2399ms
14:19:03 | ["Mind Freeze"] instant input 102ms False 463.16ms
14:19:03 | ["Mind Freeze"] instant input 84ms False 480.1028ms
14:19:01 | ["Icy Touch"] GCD: 1466ms \| Cast: 0ms \| Next spell 1113ms
14:19:01 | ["Icy Touch"] ... AfterCastAuraExpected 1900ms
14:19:01 | ["Icy Touch"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:19:01 | ["Icy Touch"] instant input 60ms True 155.0401ms
14:19:01 | ["Unbreakable Armor"] instant input 94ms False 471.0543ms
14:19:00 | ["Unbreakable Armor"] instant input 61ms False 452.7274ms
14:18:59 | Kill credit detected! Known kills: 1 \| Fighting with: 2
14:18:58 | ["Blood Boil"] GCD: 1500ms \| Cast: 0ms \| Next spell 1147ms
14:18:58 | ["Blood Boil"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:18:58 | ["Blood Boil"] instant input 89ms True 112.0772ms
14:18:58 | ["Unbreakable Armor"] instant input 77ms False 454.7638ms
14:18:58 | ["Unbreakable Armor"] instant input 94ms False 470.7531ms
14:18:56 | ["Plague Strike"] GCD: 1465ms \| Cast: 0ms \| Next spell 1112ms
14:18:56 | ["Plague Strike"] ... AfterCastAuraExpected 1900ms
14:18:56 | ["Plague Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:18:56 | ["Plague Strike"] instant input 62ms True 71.1731ms
14:18:56 | ["Unbreakable Armor"] instant input 69ms False 463.929ms
14:18:55 | ["Unbreakable Armor"] instant input 88ms False 476.6205ms
14:18:54 | ["Icy Touch"] GCD: 1500ms \| Cast: 0ms \| Next spell 1147ms
14:18:54 | ["Icy Touch"] ... AfterCastAuraExpected 1900ms
14:18:54 | ["Icy Touch"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:18:54 | ["Icy Touch"] instant input 101ms True 0.0002ms
14:18:53 | ["Unbreakable Armor"] instant input 62ms False 471.0888ms
14:18:53 | ["Unbreakable Armor"] instant input 72ms False 461.5869ms
14:18:52 | New Plan= " Combat"
14:18:52 | ["Death Grip"] GCD: 0ms \| Cast: 0ms \| Next spell -353ms
14:18:52 | ["Death Grip"] ... AfterCastWaitMeleeRange
14:18:52 | ["Death Grip"] ... AfterCastWaitCombat ? True 2633.9549ms
14:18:50 | ["Death Grip"] instant usable: True->True -- "NONE"->"CAST_SUCCESS"
14:18:50 | ["Death Grip"] instant input 98ms True 435.9363ms
14:18:49 | Stop auto interact!
14:18:49 | New Plan= " Pull Target"
14:18:46 | New Plan= " Approach Target"
14:18:46 | [NpcNameFinder] type = "None"
14:18:43 | [NpcNameFinder] type = "None"
14:18:42 | ["LocalPathingApi"] distance is close 18.544285. Keep RouteToWaypoint.
14:18:42 | New Plan= " Follow Route"
14:18:42 | ----- Corpse consumed. Total: 0 \| Remaining: 0
14:18:42 | New Plan= " Corpse Consumed"
14:18:42 | LootGoal: Loot Successful after 831.1604ms
14:18:41 | LootGoal: Target Last Target Found!
14:18:41 | New Plan= " Loot"
14:18:41 | ----- Safe to consume a corpse.
14:18:41 | New Plan= " Consume Corpse"
14:18:41 | Lost target!
14:18:41 | Kill credit detected! Known kills: 1 \| Fighting with: 0
14:18:40 | ["Blood Strike"] GCD: 1500ms \| Cast: 0ms \| Next spell 1147ms
14:18:40 | ["Blood Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:18:40 | ["Blood Strike"] instant input 59ms True 139.381ms
14:18:40 | ["Unbreakable Armor"] instant input 92ms False 475.4231ms
14:18:39 | ["Unbreakable Armor"] instant input 92ms False 473.8786ms
14:18:38 | ["Plague Strike"] GCD: 1466ms \| Cast: 0ms \| Next spell 1113ms
14:18:38 | ["Plague Strike"] ... AfterCastAuraExpected 1900ms
14:18:38 | ["Plague Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:18:38 | ["Plague Strike"] instant input 77ms True 121.2399ms
14:18:37 | ["Unbreakable Armor"] instant input 103ms False 463.9906ms
14:18:37 | ["Unbreakable Armor"] instant input 52ms False 447.1823ms
14:18:35 | ["Icy Touch"] GCD: 1466ms \| Cast: 0ms \| Next spell 1113ms
14:18:35 | ["Icy Touch"] ... AfterCastAuraExpected 1900ms
14:18:35 | ["Icy Touch"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:18:35 | ["Icy Touch"] instant input 76ms True 149.0609ms
14:18:35 | ["Unbreakable Armor"] instant input 71ms False 461.0127ms
14:18:34 | ["Unbreakable Armor"] instant input 55ms False 477.7563ms
14:18:34 | New Plan= " Combat"
14:18:34 | ["Death Grip"] GCD: 0ms \| Cast: 0ms \| Next spell -353ms
14:18:33 | ["Death Grip"] ... AfterCastWaitMeleeRange
14:18:33 | ["Death Grip"] ... AfterCastWaitCombat ? True 1966.166ms
14:18:31 | ["Death Grip"] instant usable: True->True -- "NONE"->"CAST_SUCCESS"
14:18:31 | ["Death Grip"] instant input 78ms True 153.8868ms
14:18:31 | ["Death Grip"] instant input 71ms False 460.6528ms
14:18:31 | ["Death Grip"] instant input 55ms False 478.2108ms
14:18:30 | Stop auto interact!
14:18:30 | New Plan= " Pull Target"
14:18:27 | New Plan= " Approach Target"
14:18:27 | [NpcNameFinder] type = "None"
14:18:25 | FollowRouteGoal: Random jump
14:18:23 | FollowRouteGoal: RefillWaypoints - findClosest:False - ThereAndBack:True
14:18:20 | [NpcNameFinder] type = "None"
14:18:20 | Pathfinder - 19.266893 - "<-340.39734, 1836.0862, 75.157425>" -> "<-342.51514, 1816.936, 37.428104>" 0.9495ms
14:18:20 | ["LocalPathingApi"] total distance 0 > 100. Have to clear RouteToWaypoint.
14:18:20 | New Plan= " Follow Route"
14:18:20 | ----- Corpse consumed. Total: 0 \| Remaining: 0
14:18:20 | New Plan= " Corpse Consumed"
14:18:20 | LootGoal: Loot Successful after 812.3344ms
14:18:19 | LootGoal: Target Last Target Found!
14:18:19 | New Plan= " Loot"
14:18:19 | ----- Safe to consume a corpse.
14:18:19 | New Plan= " Consume Corpse"
14:18:19 | Lost target!
14:18:19 | ["Blood Strike"] GCD: 1500ms \| Cast: 0ms \| Next spell 1147ms
14:18:19 | ["Blood Strike"] instant usable: True->False -- "NONE"->"CAST_SUCCESS"
14:18:19 | ["Blood Strike"] instant input 79ms True 120.95ms
14:18:19 | Kill credit detected! Known kills: 1 \| Fighting with: 0
14:18:18 | ["Unbreakable Armor"] instant input 102ms False 466.0244ms
14:18:18 | ["Unbreakable Armor"] instant input 98ms False 467.2526ms
14:18:16 | ["Plague Strike"] GCD: 1466ms \| Cast: 0ms \| Next spell 1113ms
14:18:16 | ["Plague Strike"] ... AfterCastAuraExpected 1900ms


</body>
</html>

**Comment by LaneBren** [comment]
[DeathKnight_80_Frost.txt](https://github.com/Xian55/WowClassicGrindBot/files/9686521/DeathKnight_80_Frost.txt)

Uploading as .txt, not sure how to share this in comment as json

**Comment by Xian55** [comment]
Hello there the related log files can be found the following locations
- `WowClassicGrindBot\BlazorServer\`
- `WowClassicGrindBot\BlazorServer\bin\Release\net6.0`
- `WowClassicGrindBot\BlazorServer\bin\x64\Release\net6.0`
- `WowClassicGrindBot\BlazorServer\bin\x86\Release\net6.0`

example file format: `out20221004.log`

**Comment by LaneBren** [comment]
[out20220929_001.log](https://github.com/Xian55/WowClassicGrindBot/files/9704847/out20220929_001.log)
 
Sorry, here is associated log file. Side note, skinning seems to be running into the same issues where the corpses aren't being searched for.

**Comment by Xian55** [comment]
About the skinning case i suggest to take a look at the following issues:

* https://github.com/Xian55/WowClassicGrindBot/issues/428
* https://github.com/Xian55/WowClassicGrindBot/pull/388
* https://github.com/Xian55/WowClassicGrindBot/issues/404

**Comment by LaneBren** [comment]
> About the skinning case i suggest to take a look at the following issues:
> 
> * [f.lux conflicts with NpcNameFinder - skinning not working #428](https://github.com/Xian55/WowClassicGrindBot/issues/428)

Confirmed not using f.lux and changed font to the one you are using from this issue thread - no fix.

> * [Feature: Rework `NpcNameFinder` - More reliable mouse targeting and skinning due `Interact with Mouseover` #388](https://github.com/Xian55/WowClassicGrindBot/pull/388)

I have cloned latest repo, doesn't seem to fix it.

> * [Retry after corpse herbing and skinning fail #404](https://github.com/Xian55/WowClassicGrindBot/issues/404)

This issue seems to be the cause for #388. Also, hotkeys are set correctly as was issue for the other user - no fix.

Another log below with different character failing at skinning and finding names to loot from. It did search once and missed the mob.

[out20221004_001.log](https://github.com/Xian55/WowClassicGrindBot/files/9711907/out20221004_001.log)

**Comment by LaneBren** [comment]
I think I have fixed the skinning at least. I guess every anti-aliasing settings matters!

Might be good to add the Advanced setting that you had posted in #428 to the readme.

**Comment by Xian55** [comment]
Personally i'm using **Anti-aliasing: Disabled**, so no pixel blur/blending happens, hard edges yields better result for **NpcNameFinder**. https://github.com/Xian55/WowClassicGrindBot/commit/23dba03776533240febe2ed4c7776d03eee7b175

Good point, nonetheless.

**Comment by Xian55** [comment]
> I think I have fixed the skinning at least. I guess every anti-aliasing settings matters!
> 
> Might be good to add the Advanced setting that you had posted in #428 to the readme.

Since last time, have you experienced such issue ?

**Comment by LaneBren** [comment]
There are still some issues with not looting when walking up to body - but probably only ~5% of the time. It logs as consumed and walks away, I'll need to run it and try to reproduce.

Other issue when searching for the corpse during skinning sequence it will find a corpse in the distance killed by another player or one that it didn't consume before. I'm guessing it is scanning top-left to bottom-right because it seems to like to find these far away corpses reliably. Not a big issue, but have noticed it happen consistently under the right circumstances. Just surprising it can easily read those names that reliably...

**Comment by Xian55** [comment]
> There are still some issues with not looting when walking up to body - but probably only ~5% of the time. It logs as consumed and walks away, I'll need to run it and try to reproduce.

Skinning/Herbing/Salvaging works best when it fights with a single mob at the time, as it take advantage of the target last target feature, so mouse action is not necessary, this one is actually a huge update introduced in wotlk pre patch. Before that it was mandatory to use mouse for Skinning or Herbing.

There are only limited capabilities to detect items/money picked up by the player during loot, when the loot is unable to picked up due inventory full, brings a lot of complexity to taken care of, not to mention client-server communication latency. On the other hand, if you play as a ranged, sometimes there are no clear straight path to the corpse due terrain elevation. 

---

> I'm guessing it is scanning top-left to bottom-right because it seems to like to find these far away corpses reliably.

It uses, the center point of the screen as reference, and checks the npc name text bounding box click point - where most likely to have something to interact with. It compares the distance and takes the shortest distance, note that the click point can be varied based on Width and Height factors.

**Comment by Xian55** [comment]
In case you were playing as hunter or warlock or any kind of class which might lose an item shortly after the target has been killed, after #454 less likely to skip loot.
